src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/CodeActions/RazorCodeActionTranslatorManager.ts

```diff
@@ -0,0 +1,38 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import * as vscode from 'vscode';
+import { IRazorCodeActionTranslator } from './IRazorCodeActionTranslator';
+
+export class RazorCodeActionTranslatorManager {
+
+    constructor(private readonly codeActionTranslators: IRazorCodeActionTranslator[]) { }
+
+    public canHandle(
+        codeAction: vscode.Command,
+        codeContext: vscode.CodeActionContext,
+        document: vscode.TextDocument,
+    ): boolean {
+        for (const actionTranslator of this.codeActionTranslators) {
+            if (actionTranslator.canHandleCodeAction(codeAction, codeContext, document)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public applyEdit(
+        uri: vscode.Uri,
+        edit: vscode.TextEdit,
+    ): [vscode.Uri?, vscode.TextEdit?] {
+        for (const actionTranslator of this.codeActionTranslators) {
+            if (actionTranslator.canHandleEdit(uri, edit)) {
+                return actionTranslator.applyEdit(uri, edit);
+            }
+        }
+        throw new Error('ApplyEdit should always be handled by one of the ActionTranslators because it should either automatically remap or they should manually fiddle with it.');
```

Comment:
Hmm I'm not sure how this is not getting hit. With how it is, when `tryApplyCodeActions()` is called, the above for loop would do nothing because the only translator we have says it can't handle the edit. I feel like I'm missing something here.

Reply:
`tryApplyingCodeActions` is only hit if we [don't get a remap response](https://github.com/dotnet/aspnetcore-tooling/pull/1466/files/5dce18afdc5c6cfc7337bcf85008d8bfa3d38baf#diff-6c27f33b4510169ba1e1fc7bbc6c70f9R56) from `mapToDocumentRange`. I guess the message here is a bit deceptive, but it's also already really long, do you have a recommendation on a message which doesn't lie and isn't super long?